assets/common/js/models/Player.js

```diff
@@ -1,3 +1,24 @@
 function Player() {
-    
+    this.score = 0;
+    this.numCharsEntered = 0;
+    this.lives = 5;
+}
+
+/**
+ * Player enters a word correctly.
+ * @param {Word} word the Word object that player entered
+ */
+Player.prototype.enterWord = function(word) {
+    // We add 1 since Space/Enter key counts as a character
+    this.numCharsEntered += word.text.length + 1;
+    this.score++;
+
+    if(word.isBonus) this.lives ++;
+}
+
+/**
+ * Player misses a word
+ */
+Player.prototype.missWord = function() {
+    // Penalize the player here
 }
```

Comment:
I think it would be better to have this in GameController.js since counting chars and updating lives makes more sense to be performed by the GameController becuase when your lives are 0 you need to call GameController.stop()

Reply:
Thank you for your comment! I feel that the responsibility of maintaining attributes that relate to player should not be imposed on the game controller. 

In our model, GameController maintains the attributes that are important for the functioning of the overall game and enforcing of game rules. We have created other models such as Player, Canvas, Dictionary to maintain variables and actions that relate to them and take burden off of the GameController (For us, it's a way of organizing our files and reduce the number of codes in the GameController). In a sense, GameController is like a referee or umpire in the real world.

In the real world, when playing a game, we normally say that "this player has 3 lives left", We tie the number of lives and personal score to the "player", not the controller.

There is a way to check if the player has 0 lives and if so, end the game from the GameController while keeping the number of lives in Player. Please look at my recent commit - **5015660**9a07621db12aa813c726b647d328d1679.

Reply:
Oh, I see. You are right. Now I have a clearer ideo of how to implement issue #16